components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsLogger.java

```diff
@@ -122,8 +159,37 @@ public void error(String value) {
         }
     }
 
+    @HostAccess.Export
+    public void error(Object... values) {
+
+        if (values != null) {
+            String resultMessage;
+            if (values.length == 0) {
+                resultMessage = "";
+            } else if (values.length == 1) {
+                resultMessage = String.valueOf(values[0]);
+            } else {
+                resultMessage = Arrays.stream(values).map(String::valueOf).collect(Collectors.joining(","));
+            }
+            logger.error(resultMessage);
+            logDiagnosticEvent("Error: " + resultMessage, DiagnosticLog.ResultStatus.FAILED);
+        }
+    }
+
     @HostAccess.Export
     public void log(String message, Object... values) {
 
     }
+
+    private void logDiagnosticEvent(String resultMessage, DiagnosticLog.ResultStatus resultStatus) {
```

Comment:
Can we use this private methods in string based log messages as well ?

Reply:
as additional improvement, that can be done